examples/FungibleToken/src/lib.rs

```diff
@@ -0,0 +1,164 @@
+        self.symbol = symbol;
+    }
+    fn symbol(&self) -> String {
+        self.symbol.clone()
+    }
+    fn total_supply(&self) -> u128 {
+        self.total_supply
+    }
+    // TODO: Use checked math and return error if Overflow.
+    fn increase_total_supply(&mut self, amount: u128) {
+        self.total_supply += amount;
+    }
+    // TODO: Use checked math and return error if Underflow.
+    fn decrease_total_supply(&mut self, amount: u128) {
+        self.total_supply -= amount;
+    }
+    fn set_balance(&mut self, account: &ProgramId, amount: u128) {
+        self.balances.insert(*account, amount);
+    }
+    fn get_balance(&self, account: &ProgramId) -> u128 {
+        *self.balances.get(account).ok_or(0u128).unwrap()
+    }
+    fn mint(&mut self, account: &ProgramId, amount: u128) {
+        unsafe {
+            FUNGIBLE_TOKEN.increase_total_supply(amount);
+            let old_balance = FUNGIBLE_TOKEN.get_balance(account);
+            // TODO: safe math for overflow detection.
+            FUNGIBLE_TOKEN.set_balance(account, old_balance + amount);
+        }
+    }
+    fn burn(&mut self, account: &ProgramId, amount: u128) {
+        unsafe {
+            FUNGIBLE_TOKEN.decrease_total_supply(amount);
+            let old_balance = FUNGIBLE_TOKEN.get_balance(account);
+            // TODO: safe math for underflow detection.
+            FUNGIBLE_TOKEN.set_balance(account, old_balance - amount);
+        }
+    }
+}
+
+#[derive(Debug, Decode, TypeInfo)]
+struct InitConfig {
+    name: String,
+    symbol: String,
+}
+
+#[derive(Debug, Decode, TypeInfo)]
+struct MintInput {
+    account: H256,
+    amount: u128,
+}
+
+#[derive(Debug, Decode, TypeInfo)]
+struct BurnInput {
+    account: H256,
+    amount: u128,
+}
+
+#[derive(Debug, Decode, TypeInfo)]
+enum Action {
+    Mint(MintInput),
+    Burn(BurnInput),
+}
+
+#[derive(Debug, Encode, TypeInfo)]
+struct TransferData {
+    from: ProgramId,
+    to: ProgramId,
+    amount: u128,
+}
+
+#[derive(Debug, Encode, TypeInfo)]
+enum Event {
+    Transfer(TransferData),
+}
+
+gstd::metadata! {
+    title: "FungibleToken",
+        init:
+            input : InitConfig,
+            output : (),
+        handle:
+            input : Action,
+            output : Event,
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn handle() {
+    let action: Action = msg::load().unwrap();
+
+    match action {
+        Action::Mint(mint_input) => {
+            if mint_input.account == H256::zero() {
+                panic!("FungibleToken: Mint to zero address.");
+            }
+            let to = ProgramId(mint_input.account.to_fixed_bytes());
+            let from = ProgramId(H256::zero().to_fixed_bytes());
+            FUNGIBLE_TOKEN.mint(&to, mint_input.amount);
+            let transfer_data = TransferData {
+                from,
+                to,
+                amount: mint_input.amount,
+            };
+            msg::reply(&Event::Transfer(transfer_data), 0, 0);
```

Comment:
I suppose, that it wouldn't for-users-only-contract, right?
So we should return available gas to the source actor.

I recommend add some constant like `const GAS_RESERVATION: u64 = 3_000_000` and return amount of gas equal `exec::gas_available() - GAS_RESERVATION` to guarantee normal end of the execution, without any gas limit exceeded.